File: src/js/0067.js
var
  fs = require('fs'),
  file = '0067.txt',
  text = fs.readFileSync(file, 'UTF-8'),
  length = {},
  solution = -Number.MAX_VALUE,
  data, i, j, row, previous;

// Process that data
data = text.split(/\r\n/g);
for (i = 0; i < data.length; i++) {
  data[i] = data[i].split(' ');
  row = data[i];
  for (j = 0; j < row.length; j++) {
    row[j] = Number(row[j]);
  }
}

// Initial values
for (i = 0; i < data.length; i++) {
  row = data[i];
  for (j = 0; j < row.length; j++) {
    length[i + ',' + j] = row[j];
  }
}

// Longest path
for (i = 0; i < data.length; i++) {
  row = data[i];
  for (j = 0; j < row.length; j++) {
    if (previous) {

      // Edges:
      length[i + ',' + j] += Math.max(
        (j - 1 >= 0) ? length[(i - 1) + ',' + (j - 1)] : -Number.MAX_VALUE,
        (j < previous.length) ? length[(i - 1) + ',' + j] : -Number.MAX_VALUE
      );

      // Sneak in solution calculation:
      solution = Math.max(length[i + ',' + j], solution);
    }
  }

  previous = row;
}

console.log(solution);

File: src/perl/0067.pl
open (FOO, "<0067.in");
my $x = 0;
my @data;
while (<FOO>) {
		@{$data[$x]} = split(/ /);
		$x++;
}
@data = reverse(@data);

foreach my $j (1..$#data) {
		foreach my $k (0..$#{$data[$j]}){
				if ($data[$j-1][$k]>$data[$j-1][$k+1]) {
						$data[$j][$k] += $data[$j-1][$k];
				} else {
						$data[$j][$k] += $data[$j-1][$k+1];
				}
		}
}

print "$data[-1][0]\n";

